apps/dashboard/src/components/workflow-editor/configure-workflow-form.tsx

```diff
@@ -0,0 +1,229 @@
+import { Input, InputField } from '../primitives/input';
+import { Separator } from '../primitives/separator';
+import { Switch } from '../primitives/switch';
+import { TagInput } from '../primitives/tag-input';
+import { Textarea } from '../primitives/textarea';
+import { configureWorkflowSchema, MAX_DESCRIPTION_LENGTH } from './schema';
+
+type ConfigureWorkflowFormProps = {
+  workflow: WorkflowResponseDto;
+  debouncedUpdate: (data: UpdateWorkflowDto) => void;
+  patch: (data: PatchWorkflowDto) => void;
+};
+
+export const ConfigureWorkflowForm = (props: ConfigureWorkflowFormProps) => {
+  const { workflow, debouncedUpdate, patch } = props;
+  const isReadOnly = workflow.origin === WorkflowOriginEnum.EXTERNAL;
+  const [isPauseModalOpen, setIsPauseModalOpen] = useState(false);
+  const tagsQuery = useTagsQuery();
+
+  const form = useForm({
+    defaultValues: {
+      name: workflow.name,
+      workflowId: workflow.workflowId,
+      description: workflow.description,
+      tags: workflow.tags,
+    },
+    resolver: zodResolver(configureWorkflowSchema),
+  });
+
+  const statusForm = useForm({
+    defaultValues: {
+      active: workflow.active,
+    },
+  });
+
+  const onPauseWorkflow = () => {
+    statusForm.setValue('active', false, { shouldValidate: true, shouldDirty: true });
+  };
+
+  useFormAutosave(form, (data) => {
+    debouncedUpdate({ ...workflow, ...data });
+  });
+
+  useFormAutosave(statusForm, (data) => {
+    patch(data);
+  });
+
+  // const setIssuesFromWorkflow = (workflow: WorkflowResponseDto) => {
+  //   const issues = flattenIssues(workflow.issues);
+  //   Object.entries(issues).forEach(([key, value]) => {
+  //     setError(key as string, { message: value }, { shouldFocus: true });
+  //   });
+  // };
+
+  // //set issues from step
+  // useEffect(() => {
+  //   setIssuesFromStep(step);
+  //   form.reset(merge(buildDefaultValues(step.controls.uiSchema ?? {}), step?.controls.values));
+  // }, [step]);
+
+  return (
+    <>
+      <ConfirmationModal
+        open={isPauseModalOpen}
+        onOpenChange={setIsPauseModalOpen}
+        onConfirm={() => {
+          onPauseWorkflow();
+          setIsPauseModalOpen(false);
+        }}
+        title={PAUSE_MODAL_TITLE}
+        description={<PauseModalDescription workflowName={workflow.name} />}
+        confirmButtonText="Proceed"
+      />
+      <PageMeta title={workflow.name} />
+      <motion.div
+        className={cn('relative flex h-full w-full flex-col')}
+        initial={{ opacity: 0 }}
+        animate={{ opacity: 1 }}
+        exit={{ opacity: 0.1 }}
+        transition={{ duration: 0.1 }}
+      >
+        <SidebarHeader className="items-center text-sm font-medium">
+          <div className="flex items-center gap-1">
+            <RouteFill />
+            <span>Configure workflow</span>
+          </div>
+        </SidebarHeader>
+        <Separator />
+        <form>
```

Comment:
The `<form/>` should be rendered by the `<Form/>` primitive component. See how they use it [here](https://react-hook-form.com/docs/formprovider). That is wrap `form` with the Form Provider.